File: matchmaker/app/src/test/java/ru/ifmo/se/dating/matchmaker/api/HttpMonitoringApiTest.java
package ru.ifmo.se.dating.matchmaker.api;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.web.client.TestRestTemplate;
import ru.ifmo.se.dating.matchmaker.MatchmakerTestSuite;

public final class HttpMonitoringApiTest extends MatchmakerTestSuite {
    @Autowired
    private TestRestTemplate rest;

    @Test
    public void healthcheck() {
        assertEquals(getHealthcheck(), "public");
    }

    private String getHealthcheck() {
        final var url = "http://localhost:8080/api/v1/monitoring/healthcheck";
        final var response = rest.getForEntity(url, String.class);
        return response.getBody();
    }
}

File: backend/matchmaker/src/test/kotlin/ru/ifmo/se/dating/matchmaker/api/HttpMonitoringApiTest.kt
package ru.ifmo.se.dating.matchmaker.api

import kotlinx.coroutines.reactor.awaitSingle
import kotlinx.coroutines.runBlocking
import org.junit.Assert
import org.junit.Test
import ru.ifmo.se.dating.matchmaker.MatchmakerTestSuite

class HttpMonitoringApiTest : MatchmakerTestSuite() {
    @Test
    fun emptyTest() = Unit

    @Test
    fun healthcheck(): Unit = runBlocking {
        Assert.assertEquals("pong", getHealthcheck())
    }

    private suspend fun getHealthcheck(): String = self
        .get()
        .uri("monitoring/healthcheck")
        .retrieve()
        .toEntity(String::class.java)
        .awaitSingle()
        .body!!
}
